lib/mongoose-store.js

```diff
@@ -29,15 +56,15 @@ Store.prototype.init = function(data) {
   var self = this;
   return Promise.map(data, function(entry) {
     var record = new self.model(entry);
-    return record.save();
+    return record.save(errorHandler(entry, 'init'));
   });
 };
 
 Store.prototype.isPersistent = true;
 
 Store.prototype.create = function(object) {
   var record = new this.model(object);
-  return record.save();
+  return record.save(errorHandler(entry, 'create'));
```

Comment:
We should always avoid implicit usage as this is hard to maintain.
Let's wrap save method (replace prototype). This will enable validation across the board.